### ISSUE TITLE
Setting up AWS Cognito

### ISSUE THREAD

**Issue by jermnelson** [LD4P/sinopia_server#25]
Setup a running Cognito instance for experimentation using Terraform.

**Comment by atz** [comment]
There are 8 different Cognito components and usage can get reasonably complex, including lambdas, email verification templates, etc.  For example, see the diagram from one 3rd party module [here](https://github.com/squidfunk/terraform-aws-cognito-auth).

I do not have a strong understanding of our expected use cases here, other than "store basic user account", "allow authentication", and "provide token".  So I don't know how much of the other stuff is in play:
- Do we allow user signup?
  - Do we require email verification? or link to Amazon/Facebook/Google accts?
    - Do we want to link to Github accounts?  That is not available by default, but seems possible using a 3rd party [wrapper/shim app](https://github.com/TimothyJones/github-cognito-openid-wrapper).
  - Do we require further manual approval?

Basically, I could use a better description of what kind of flow we want to support.

**Comment by jermnelson** [comment]
From what I recall from discussions last summer/fall, we would allow user signup who would then need to manually approved by @michelleif , at least for the MVP.  This scenario is not scalable so giving  Michelle options about what use cases Cognito can support and then let her decide or take the options to the user group for their consideration should help us to decide what components we need for this work-cycle.

**Comment by atz** [comment]
The default configurable social sign-ins are Amazon, Facebook and Google.  Any other social link requires extra infrastructure (for us to build/maintain), so should be a deliberate commitment, not just a preference.  

My understanding is that we will be using the auth token between our front end and our server, as described in ["Access Your Server-side Resources with a User Pool"](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-scenarios.html#scenario-backend).  

IF we want to allow a user who is a member of multiple groups to select which group is "active" for given operation, we will need to build that capability into the front end, and acquire the cognito auth token as described [here](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-user-groups.html).

Otherwise, we should define our groups w/ appropriate `precedence` values.  I.E., admin precedes PCC precedes institution precedes unaffiliated.

**Comment by jermnelson** [comment]
Closed, AWS setup is in Terraform.
